wallet/wallet.c

```diff
@@ -1251,7 +1251,16 @@ void wallet_invoice_save(struct wallet *wallet, struct invoice *inv)
 	 * that string for sql injections... */
 	if (!inv->id) {
 		stmt = db_prepare(wallet->db,
-			"INSERT INTO invoices (payment_hash, payment_key, state, msatoshi, label, expiry_time, pay_index) VALUES (?, ?, ?, ?, ?, ?, ?);");
+				  "INSERT INTO invoices ("
+				  " payment_hash,"
+				  " payment_key,"
+				  " state,"
+				  " msatoshi,"
+				  " label,"
+				  " expiry_time,"
+				  " pay_index,"
+				  " msatoshi_received"
```

Comment:
Still not sufficient... we should really update this line to include `msatoshi_received`: https://github.com/ElementsProject/lightning/pull/585/commits/3ea37c3ea715230e722866299086c1aea4a08890#diff-2ffd2a734c826343043d51802d347a09R1294

Indeed the latter line is more important since `msatoshi_received` is correctly set only after the invoice is resolved, and the invoice is resolved only after it is actually created and exists in the database.

Reply:
Right, added it to the update as well.